lib/iris/tests/unit/cube/test_Cube__aggregated_by.py

```diff
@@ -0,0 +1,829 @@
+                ]
+            ),
+        )
+
+
+class Test_aggregated_by__climatology(tests.IrisTest):
+    def setUp(self):
+        self.data = np.arange(100).reshape(20, 5)
+        self.aggregator = iris.analysis.MEAN
+
+    def get_result(
+        self,
+        transpose: bool = False,
+        second_categorised: bool = False,
+        bounds: bool = False,
+        partially_aligned: bool = False,
+        partially_aligned_timelike: bool = False,
+        invalid_units: bool = False,
+        already_climatological: bool = False,
+        climatological_op: bool = True,
+    ) -> Cube:
+        cube_data = self.data
+        if transpose:
+            cube_data = cube_data.T
+            axes = [1, 0]
+        else:
+            axes = [0, 1]
+        if not invalid_units:
+            units = Unit("days since 1970-01-01")
+        else:
+            units = Unit("m")
+        if partially_aligned_timelike:
+            pa_units = Unit("days since 1970-01-01")
+        else:
+            pa_units = Unit("m")
+
+        # DimCoords
+        aligned_coord = DimCoord(
+            np.arange(20),
+            long_name="aligned",
+            units=units,
+        )
+        orthogonal_coord = DimCoord(np.arange(5), long_name="orth")
+
+        if bounds:
+            aligned_coord.guess_bounds()
+
+        aligned_coord.climatological = already_climatological
+
+        dim_coords_and_dims = zip([aligned_coord, orthogonal_coord], axes)
+
+        # AuxCoords
+        categorised_coord1 = AuxCoord(
+            np.tile([0, 1], 10), long_name="cat1", units=Unit("month")
+        )
+
+        if second_categorised:
+            categorised_coord2 = AuxCoord(
+                np.tile([0, 1, 2, 3, 4], 4), long_name="cat2"
+            )
+            categorised_coords = [categorised_coord1, categorised_coord2]
+        else:
+            categorised_coords = categorised_coord1
+
+        aux_coords_and_dims = [
+            (categorised_coord1, axes[0]),
+        ]
+
+        if second_categorised:
+            aux_coords_and_dims.append((categorised_coord2, axes[0]))
+
+        if partially_aligned:
+            partially_aligned_coord = AuxCoord(
+                cube_data + 1,
+                long_name="part_aligned",
+                units=pa_units,
+            )
+            aux_coords_and_dims.append((partially_aligned_coord, (0, 1)))
+
+        # Build cube
+        in_cube = iris.cube.Cube(
+            cube_data,
+            long_name="wibble",
+            dim_coords_and_dims=dim_coords_and_dims,
+            aux_coords_and_dims=aux_coords_and_dims,
+        )
+
+        out_cube = in_cube.aggregated_by(
+            categorised_coords,
+            self.aggregator,
+            climatological=climatological_op,
+        )
+
+        return out_cube
+
+    def test_basic(self):
+        # Check the least complicated version works (set climatological, set
+        # points correctly).
```

Comment:
Now that we are using `pytest` as the runner (#4734) the test paths will appear in the output regardless of whether the tests have docstrings.  So you could change these back to docstrings if you want.  I have no opinion either way.

Reply:
Done